src/App.tsx

```diff
@@ -8,7 +8,54 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
+import { Category } from './types/enums/Category';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const [query, setQuery] = useState<string>('');
+  const [category, setCategory] = useState<string>(Category.All);
+
+  // eslint-disable-next-line prettier/prettier
+  const [selectedTodoId, setSelectedTodoId] = useState<number | null>(null);
+
+  useEffect(() => {
+    setIsLoading(true);
+
+    const getTodoList = async () => {
+      try {
+        const data = await getTodos();
+
+        setTodos(data);
+        setIsLoading(false);
+      } catch (err) {
+        setError('GET getTodoList() ERROR');
+        setIsLoading(true);
```

Comment:
> In the catch block, `setIsLoading(true);` is incorrect. It should be `setIsLoading(false);` to stop the loading state when an error occurs.

Good point. To avoid such errors and duplications remove loaders in the finally block